avoidance/src/common.cpp

```diff
@@ -8,24 +8,90 @@
 
 namespace avoidance {
 
+bool pointInsideFOV(const std::vector<FOV>& fov_vec, const PolarPoint& p_pol) {
+  for (auto fov : fov_vec) {
+    if (pointInsideFOV(fov, p_pol)) {
+      return true;
+    }
+  }
+  return false;
+}
+
 bool pointInsideFOV(const FOV& fov, const PolarPoint& p_pol) {
-  return p_pol.z <= wrapAngleToPlusMinus180(fov.azimuth_deg + fov.h_fov_deg / 2.f) &&
-         p_pol.z >= wrapAngleToPlusMinus180(fov.azimuth_deg - fov.h_fov_deg / 2.f) &&
-         p_pol.e <= fov.elevation_deg + fov.v_fov_deg / 2.f && p_pol.e >= fov.elevation_deg - fov.v_fov_deg / 2.f;
+  return p_pol.z <= wrapAngleToPlusMinus180(fov.yaw_deg + fov.h_fov_deg / 2.f) &&
+         p_pol.z >= wrapAngleToPlusMinus180(fov.yaw_deg - fov.h_fov_deg / 2.f) &&
+         p_pol.e <= fov.pitch_deg + fov.v_fov_deg / 2.f && p_pol.e >= fov.pitch_deg - fov.v_fov_deg / 2.f;
+}
+
+bool isInWhichFOV(const std::vector<FOV>& fov_vec, const PolarPoint& p_pol, int& idx) {
+  bool retval = false;
+  idx = -1;
+  for (int i = 0; i < fov_vec.size(); ++i) {
+    if (pointInsideFOV(fov_vec[i], p_pol)) {
+      if (retval) {  // if it's been found before, return false!
+        idx = -1;
+        return false;
+      }
+      idx = i;
+      retval = true;
+    }
+  }
+  return retval;
+}
+
+bool isOnEdgeOfFOV(const std::vector<FOV>& fov_vec, const PolarPoint& p_pol, int& idx) {
+  idx = -1;
+  bool retval = false;
+  if (isInWhichFOV(fov_vec, p_pol, idx)) {
+    PolarPoint just_outside = p_pol;
+    // todo: check for pitch!
+    if (wrapAngleToPlusMinus180(p_pol.z - fov_vec[idx].yaw_deg) > 0.0f) {  // to the right
+      just_outside.z = wrapAngleToPlusMinus180(fov_vec[idx].yaw_deg + fov_vec[idx].h_fov_deg / 2.0f + 1.0f);
```

Comment:
could you explain the number 1.0f? Why does being 1 deg outside have any significant meaning in contrast to being bigger than the exact edge or for example 5 degrees?

Reply:
Yeah, that's an arbitrary magic number, which is nasty. I couldn't think of a better way to do this. (Can you?)
I'll add a comment in the meantime and spend some time thinking of a better way to do this

Reply:
I tried adding FLT_MIN and FLT_EPSILON, but that doesn't work, because those constants are only significant around zero and get rounded away for larger numbers due to the inherent nonuniform distribution of floating point numbers.

I could tie it to ALPHA_RES somehow, since that somewhat meaningfully represents the smallest angle increment we care about? Any ideas?

Reply:
I like the idea of a minimum FOV being ALPHA_RES + eps.

Reply:
downside is I have to include histogram.h in common.h. I hate introducing these dependencies, but it's equally ugly to re-define that resolution in another place

Reply:
yes that sounds better, if you exchange the 1.0 by ALPHA_RES/2  that would kinda make sense